extensions/quickfile/src/balances.tsx

```diff
@@ -0,0 +1,94 @@
+import { ActionPanel, List, Action, getPreferenceValues } from "@raycast/api";
+
+import { bankGetAccounts, bankGetAccountTransactions, BankType } from "./api/bank";
+import { systemGetAccountDetails } from "./api/system";
+import { accountPath, iconPath } from "./api/urls";
+
+import { formatCurrency } from "./utils/formatting";
+import { useAsyncData } from "./utils/hooks";
+import { unique } from "./utils/data";
+import AccountDetail from "./views/AccountDetail";
+
+const bankTypePriority: Record<BankType, number> = {
+  CURRENT: 100,
+  PETTY: 80,
+  BUILDINGSOC: 70,
+  CREDITCARD: 60,
+  LOAN: 50,
+  MERCHANT: 30,
+  RESERVE: 20,
+  EQUITY: 10,
+};
+
+const bankTypeNames: Record<BankType, string> = {
+  CURRENT: "Current accounts",
+  PETTY: "Petty cash",
+  BUILDINGSOC: "Building societies",
+  LOAN: "Loan",
+  MERCHANT: "Merchant account",
+  EQUITY: "Equity",
+  CREDITCARD: "Credit cards",
+  RESERVE: "Reserve",
+};
+
+export default function Command() {
+  const { accountNumber } = getPreferenceValues();
+
+  const { data: domain, loading: loadingDomain } = useAsyncData(accountNumber && "slug", () =>
+    systemGetAccountDetails(accountNumber, ["ClientDomain"]).then((r) => r.ClientDomain)
+  );
+
+  const { data: accounts = [], loading: loadingAccounts } = useAsyncData(accountNumber && "accounts", () =>
+    bankGetAccounts()
+  );
+
+  const bankCodes = accounts.map((a) => a.NominalCode);
+  const { data: transactions = [], loading: loadingTransactions } = useAsyncData(bankCodes, (codes) =>
+    Promise.all(codes.map((code) => bankGetAccountTransactions(code, { number: 20 })))
+  );
+
+  const accountTypes = unique(accounts.map((a) => a.BankType)).sort(
+    (a, b) => (bankTypePriority[b] || 0) - (bankTypePriority[a] || 0)
+  );
+
+  return (
+    <List isLoading={loadingDomain || loadingAccounts || loadingTransactions}>
```

Comment:
Could you set the `isLoading` to `false` when the error occurs, this way we see the `No Results` screen instead.

![CleanShot 2022-05-24 at 05 09 56@2x](https://user-images.githubusercontent.com/38033107/169943188-32c0b2ab-a48f-4cd1-8921-42f0c7102065.png)